pkg/cri/server/container_create_windows_test.go

```diff
@@ -195,3 +196,51 @@ func TestMountNamedPipe(t *testing.T) {
 	specCheck(t, testID, testSandboxID, testPid, spec)
 	checkMount(t, spec.Mounts, `\\.\pipe\foo`, `\\.\pipe\foo`, "", []string{"rw"}, nil)
 }
+
+func TestHostProcessRequirements(t *testing.T) {
+	testID := "test-id"
+	testSandboxID := "sandbox-id"
+	testContainerName := "container-name"
+	testPid := uint32(1234)
+	containerConfig, sandboxConfig, imageConfig, _ := getCreateContainerTestData()
+	ociRuntime := config.Runtime{}
+	c := newTestCRIService()
+	for desc, test := range map[string]struct {
+		containerHostProcess bool
+		sandboxHostProcess   bool
+		expectError          bool
+	}{
+		"hostprocess container in non-hostprocess sandbox should fail": {
+			containerHostProcess: true,
+			sandboxHostProcess:   false,
+			expectError:          true,
+		},
+		"hostprocess container in hostprocess sandbox should be fine": {
+			containerHostProcess: true,
+			sandboxHostProcess:   true,
+			expectError:          false,
+		},
+		"non-hostprocess container in hostprocess sandbox should fail": {
+			containerHostProcess: false,
+			sandboxHostProcess:   true,
+			expectError:          true,
+		},
+		"non-hostprocess container in non-hostprocess sandbox should be fine": {
+			containerHostProcess: false,
+			sandboxHostProcess:   false,
+			expectError:          false,
+		},
+	} {
+		t.Logf("TestCase %q", desc)
+		containerConfig.Windows.SecurityContext.HostProcess = test.containerHostProcess
+		sandboxConfig.Windows.SecurityContext = &runtime.WindowsSandboxSecurityContext{
+			HostProcess: test.sandboxHostProcess,
+		}
+		_, err := c.containerSpec(testID, testSandboxID, testPid, "", testContainerName, testImageName, containerConfig, sandboxConfig, imageConfig, nil, ociRuntime)
+		if test.expectError {
+			assert.Error(t, err)
+		} else {
+			assert.NoError(t, err)
+		}
```

Comment:
This is a good opportunity to use subtests for the table rather than logging the test case separately; that'll allow the output of `go test` to tell you which subtest(s) failed while continuing to execute and show passing output for the ones that pass.

```suggestion
		t.Run(desc, func(t *testing.T) {
			containerConfig.Windows.SecurityContext.HostProcess = test.containerHostProcess
			sandboxConfig.Windows.SecurityContext = &runtime.WindowsSandboxSecurityContext{
				HostProcess: test.sandboxHostProcess,
			}
			_, err := c.containerSpec(testID, testSandboxID, testPid, "", testContainerName, testImageName, containerConfig, sandboxConfig, imageConfig, nil, ociRuntime)
			if test.expectError {
				assert.Error(t, err)
			} else {
				assert.NoError(t, err)
			}
		})	
		```

Reply:
Great idea, changing shortly

Reply:
Done!